### ISSUE TITLE
Boolean/Check-box state is incorrect on full table-block view

### ISSUE THREAD

**Issue by ttben** [Budibase/budibase#9151]
**Hosting**
- Self
   - Budibase Version: 2.2.3

Data : tinyint(1) from MySQL database, mapped as Boolean in budiBase

**Describe the bug**
When displaying a table-block with a column displaying a boolean, the overall view shows a "true/ticked checkbox" state, even if the actual state is false.
![Screenshot from 2022-12-22 13-46-59](https://user-images.githubusercontent.com/6167712/209140442-df9e1900-0291-43e0-85b8-40d389e7490d.png)

The side panel correctly reflects the actual state of the value (the one stored in the data source)

![Screenshot from 2022-12-22 14-01-15](https://user-images.githubusercontent.com/6167712/209140555-20a2371e-414e-48b6-a8f7-a7919eb5ce23.png)

**Comment by melohagan** [comment]
Hey @ttben 

I tried to reproduce this, but I didn't see any issue with `tinyint(1)`.

Could you please provide reproduction steps, and if possible an app export please?

**Comment by ttben** [comment]
Hi! (and happy new year :confetti_ball: )

I tried, but failed to reproduce the issue on a tiny (haha) example. Here is the steps I followed :

- Create a minimal schema with a `tinyint` column 
```
CREATE TABLE `test-repro-bug-budibase-9151` (
  `thisisatinyintone` tinyint(1) NOT NULL,
  `uuid` bigint(20) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

ALTER TABLE `test-repro-bug-budibase-9151`
  ADD PRIMARY KEY (`uuid`);
COMMIT;
```

- Add it as datasource
- Update the column to be a `Boolean` and not a `Number`
- Generate the CRUD views
- Add data in database directly with `0` and `1` 
- The status is correctly reflected in the _datasource_ view in budibase
- The status is correctly reflected in the production website in the table view
- The status is correctly reflected in the production website in the edit pane view

**BUT**, when going back to my actual database and views to check what I did differently, **_I noticed that in budibase datasource view I see the correct state, but in the production website the state is wrong._** 

Database value 
![image](https://user-images.githubusercontent.com/6167712/211260082-9cde6240-332c-4fa0-ae81-9fe0572d8e1f.png)

Datasource view (correct, ie matches DB state)
![image](https://user-images.githubusercontent.com/6167712/211260021-6ef008be-1adf-4534-a4bd-d460b92f97b9.png)

Website **table** (incorrect)
![image](https://user-images.githubusercontent.com/6167712/211259966-9085bc43-24e7-4a97-b846-7ee50a44a861.png)

Website edit pane (correct)
![image](https://user-images.githubusercontent.com/6167712/211260206-ce85ec2d-4edd-4ff1-901b-0be899ee93b6.png)

The issue appears in a generated table block in a generated screen and I didnt touch anything.

SO, **when recreating** the whole screen, the issue disappears. I will go this way then ... I was lucky enough to be able to trash this auto-generated screen and recreate another one!
